### PR TITLE
40139 vaos breadcrumb replacement

### PR DESCRIPTION
**Pull request by ryanshaw**
## Description
This PR replaces the React breadcrumb component with the newer [breadcrumb web component](https://design.va.gov/components/breadcrumbs) from the VA.gov design system.

## Original issue(s)
[department-of-veterans-affairs/va.gov-team/issues/40139](https://github.com/department-of-veterans-affairs/va.gov-team/issues/40139)

## Testing done
- Cypress e2e tests
- Unit testing

## Notes

- Previously the `Past` and `Pending` breadcrumb status was stored in the Redux store, all the other breadcrumbs were added to the React component as `children`
- The new web component only accepts `children` and strings or functions passed as props
- This led to some minor changes in the URL structure. Previously appointment detail routes were `/va/:appointmentId` and `/cc/:appointmentId` regardless of if the appointment was past or pending, etc. I've added an optional param to those routes that the new breadcrumb component uses to determine the display of past and pending breadcrumbs. That means that an appointment could have the following URLs associated with it and still display:
  - `/va/:appointmentId`
  - `/cc/:appointmentId`
  - `/past/va/:appointmentId`
  - `/past/cc/:appointmentId`
  - `/pending/va/:appointmentId`
  - `/pending/cc/:appointmentId`

## Questions

- ~~Is the URL options an issue?~~ ([see comment](https://github.com/department-of-veterans-affairs/vets-website/pull/21496#issuecomment-1176340402))
- Is there a better way to approach this? It simplified things a lot and made the breadcrumbs less dependent on state, etc
- Because of the way the `Past` and `Present` links work within [AppointmentListNavigation.jsx](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/applications/vaos/appointment-list/components/AppointmentListNavigation.jsx) the web component is not re-rendering -- maybe there is a better way to handle that?

## Acceptance criteria
- [ ] 

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Comment by ryanshaw** [comment]
For reference, here's the spacing without the class. I believe the class was from when the breadcrumbs were created specifically for and within the VAOS app -- so I've removed it. 

image

File: src/applications/vaos/appointment-list/components/AppointmentsPageV2/AppointmentListItem.jsx [hunk 1]
```diff
@@ -13,7 +13,6 @@ import {
   VIDEO_TYPES,
   SPACE_BAR,
 } from '../../../utils/constants';
-import { updateBreadcrumb } from '../../redux/actions';
 import { selectFeatureStatusImprovement } from '../../../redux/selectors';
 
 function VideoAppointmentDescription({ appointment }) {
```
Comment (vbahinwillit) [inline src/applications/vaos/appointment-list/components/AppointmentsPageV2/AppointmentListItem.jsx:18]: Small but I would add PropTypes. We've been trying to get rid of the easy eslint warning when we touch files.

File: src/applications/vaos/appointment-list/components/AppointmentsPageV2/RequestListItem.jsx [hunk 2]
```diff
@@ -2,34 +2,26 @@ import React from 'react';
...
-}) {
+function handleClick({ history, link, idClickable, featureStatusImprovement }) {
   return () => {
     if (!window.getSelection().toString()) {
       focusElement(`#${idClickable}`);
       history.push(link);
 
       if (featureStatusImprovement) {
-        dispatch(updateBreadcrumb({ title: 'Pending', path: '/pending' }));
+        //  dispatch(updatecrumb({ title: 'Pending', path: '/pending' }));
```
Comment (vbahinwillit) [inline src/applications/vaos/appointment-list/components/AppointmentsPageV2/RequestListItem.jsx:18]: I would remove this.

**Review by vbahinwillit — COMMENTED** [review]
LGTM!

**Comment by vbahinwillit** [comment]
I would also follow up with Peter about the URL changes.

**Comment by ryanshaw** [comment]
> I would also follow up with Peter about the URL changes.

Thanks, I will do that. 

I also need to clean [RequestListItem.jsx](https://github.com/department-of-veterans-affairs/vets-website/blob/40139-vaos-breadcrumb-replacement/src/applications/vaos/appointment-list/components/AppointmentsPageV2/RequestListItem.jsx) up a bit, I left some comments in there.

**Comment by ryanshaw** [comment]
@outerpress Is there a UX issue with the link/route changes I've made in this PR? It's outline in the notes section.

File: src/applications/vaos/appointment-list/components/AppointmentsPageV2/AppointmentListItem.jsx (hunk 1 above)
Comment (ryanshaw) [reply src/applications/vaos/appointment-list/components/AppointmentsPageV2/AppointmentListItem.jsx:18]: Will do, was wondering about that.

**Review by simiadebowale — COMMENTED** [review]
LGTM, I would just add propTypes to the AppointmentListItem component.

**Review by vbahinwillit — APPROVED** [review]
LGTM!

**Pull request merged**